File: Javascript/src/01.AlgorithmicToolbox/Week3GreedyAlgorithms/moneyChange.js
/**
 * Problem 1 - Money Change
 * @param {number} value total value of coins
 * @returns {number} minimum number of coins of value 1, 5 and 10 that exactly equal value
 */
export function minCoins(value) {
  const numberOf10 = Math.floor(value / 10);
  value %= 10;
  const numberOf5 = Math.floor(value / 5);
  value %= 5;
  const numberOf1 = value;
  return numberOf1 + numberOf5 + numberOf10;
}

File: Rust/src/_01_algorithmic_toolbox/week3_greedy_algorithms/money_change.rs
/// Problem 1 - Money Change
///
/// Returns the minimum number of 1, 5 and 10 valued coins that make up a given input `value`
///
/// * `value` the total value of the coins
pub fn money_change(value: u64) -> u64 {
    let number_of_10 = value / 10;
    let value = value % 10;
    let number_of_5 = value / 5;
    let value = value % 5;
    let number_of_1 = value;
    return number_of_10 + number_of_5 + number_of_1;
}

#[cfg(test)]
mod test {
    use super::*;

    #[test]
    fn test_returns_minimum_amount_of_coins() {
        assert_eq!(money_change(2), 2);
        assert_eq!(money_change(28), 6);
    }
}
